.github/workflows/test.yml

```diff
@@ -170,7 +170,7 @@ jobs:
       matrix:
         os: [ macos-latest, windows-latest, ubuntu-latest ]
     env:
-      JAVA_TOOL_OPTIONS: -XX:+TieredCompilation -XX:TieredStopAtLevel=1
+      JAVA_TOOL_OPTIONS: -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64M
```

Comment:
What is the default?

Reply:
as it would be to easy otherwise it varies depending on the platform between 32-48MB :D 
https://chriswhocodes.com/hotspot_options_openjdk17.html

In the failed runs the 48MB limit was hit.

Reply:
Crazy

Reply:
Lol this page is so nineties

Reply:
It's a little strange though that we need more code cache, and only on macOS. But I guess we don't really support production deployments on macOS so :shrug:

Reply:
@npepinpe it's actually failing during compilation hitting the code cache limit on macos

Reply:
I know, I'm still wondering why it happens and if it implies anything about our code in the end, but I also get it's not all that important to dig into as it probably wouldn't affect any users